chips/nrf52/src/i2c.rs

```diff
@@ -0,0 +1,446 @@
+//! Implementation of I2C for nRF52 using EasyDMA.
+//!
+//! This module supports nRF52's two I2C master (`TWIM`) peripherals,
+//! but not I2C slave (`TWIS`).
+//!
+//! - Author: Jay Kickliter
+//! - Author: Andrew Thompson
+//! - Date: Nov 4, 2017
+
+use core::cell::Cell;
+use kernel::common::take_cell::TakeCell;
+use kernel::hil;
+use nrf5x::nvic;
+use nrf5x::peripheral_interrupts::NvicIdx;
+use nrf5x::pinmux::Pinmux;
+
+/// An I2C master device.
+///
+/// A `TWIM` instance wraps a `registers::TWIM` together with
+/// addition data necessary to implement an asynchronous interface.
+pub struct TWIM {
+    registers: *const registers::TWIM,
+    nvic_idx: NvicIdx,
```

Comment:
NVIC management should go away when we merge #687. If this gets merged first, I'll handle that no problem, but might need a bit of help verifying I don't break anything. If #687 is merged first... well basically you get to just assume that the nvic is enabled...